### PR TITLE
Fixes security dependencies in Go

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR fixes the dependency alerts listed here: https://github.com/camunda-cloud/zeebe/security/dependabot/clients/go/go.sum/github.com%2Fopencontainers%2Frunc/open and https://github.com/camunda-cloud/zeebe/security/dependabot/clients/go/go.sum/github.com%2Fopencontainers%2Fimage-spec/open. The first go around it seems doing a tidy removed the dependencies, such that we didn't actually fix them. The important bits to verify is that the dependencies listed in the alerts are updated to the minimum versions in the `go.sum` file.

As these are test only dependencies, I don't see a need to back port this.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: clients/go/go.sum [hunk 1]
```diff
@@ -577,15 +580,17 @@ github.com/opencontainers/go-digest v1.0.0-rc1.0.20180430190053-c9281466c8b2/go.
...
 github.com/opencontainers/image-spec v1.0.1/go.mod h1:BtxoFyWECRxE4U/7sNtV5W15zMzWCbyJoFRP3s7yZA0=
+github.com/opencontainers/image-spec v1.0.2 h1:9yCKha/T5XdGtO0q9Q9a6T5NUCsTn/DrBg0D7ufOcFM=
+github.com/opencontainers/image-spec v1.0.2/go.mod h1:BtxoFyWECRxE4U/7sNtV5W15zMzWCbyJoFRP3s7yZA0=
 github.com/opencontainers/runc v0.0.0-20190115041553-12f6a991201f/go.mod h1:qT5XzbpPznkRYVz/mWwUaVBUv2rmF59PVA73FjuZG0U=
 github.com/opencontainers/runc v0.1.1/go.mod h1:qT5XzbpPznkRYVz/mWwUaVBUv2rmF59PVA73FjuZG0U=
 github.com/opencontainers/runc v1.0.0-rc8.0.20190926000215-3e425f80a8c9/go.mod h1:qT5XzbpPznkRYVz/mWwUaVBUv2rmF59PVA73FjuZG0U=
 github.com/opencontainers/runc v1.0.0-rc9/go.mod h1:qT5XzbpPznkRYVz/mWwUaVBUv2rmF59PVA73FjuZG0U=
 github.com/opencontainers/runc v1.0.0-rc93/go.mod h1:3NOsor4w32B2tC0Zbl8Knk4Wg84SM2ImC1fxBuqJ/H0=
-github.com/opencontainers/runc v1.0.2 h1:opHZMaswlyxz1OuGpBE53Dwe4/xF7EZTY0A2L/FpCOg=
 github.com/opencontainers/runc v1.0.2/go.mod h1:aTaHFFwQXuA71CiyxOdFFIorAoemI04suvGRQFzWTD0=
```
Comment (oleschoenburg) [inline clients/go/go.sum:591]: :question: Same concern as above, this looks like we are still including a vulnerable version of runc?

File: clients/go/go.sum [hunk 2]
```diff
@@ -577,15 +580,17 @@ github.com/opencontainers/go-digest v1.0.0-rc1.0.20180430190053-c9281466c8b2/go.
 github.com/opencontainers/go-digest v1.0.0 h1:apOUWs51W5PlhuyGyz9FCeeBIOUDA/6nW8Oi/yOhh5U=
 github.com/opencontainers/go-digest v1.0.0/go.mod h1:0JzlMkj0TRzQZfJkVvzbP0HBR3IKzErnv2BNG4W4MAM=
 github.com/opencontainers/image-spec v1.0.0/go.mod h1:BtxoFyWECRxE4U/7sNtV5W15zMzWCbyJoFRP3s7yZA0=
-github.com/opencontainers/image-spec v1.0.1 h1:JMemWkRwHx4Zj+fVxWoMCFm/8sYGGrUVojFA6h/TRcI=
 github.com/opencontainers/image-spec v1.0.1/go.mod h1:BtxoFyWECRxE4U/7sNtV5W15zMzWCbyJoFRP3s7yZA0=
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply clients/go/go.sum:583]: To be perfectly honest, I'm also not sure. It seems pretty common, but I guess now's a good time to learn why it does that :upside_down_face:

**Comment by npepinpe** [comment]
Regarding the multiple versions included in the go.sum file, the [documentation for the file](https://go.dev/ref/mod#go-sum-files) has some info about this:

> The go.sum file may contain hashes for multiple versions of a module. The go command may need to load go.mod files from multiple versions of a dependency in order to perform minimal version selection. go.sum may also contain hashes for module versions that aren’t needed anymore (for example, after an upgrade). go mod tidy will add missing hashes and will remove unnecessary hashes from go.sum.

In our case, running `go mod tidy` still doesn't remove all the unnecessary hashes. I presume this is because of the algorithm used to perform minimum version selection. It seems on Go 1.17 this issue is changed somewhat and only the minimum required dependencies are kept in the go.sum and go.mod files, but we're still on Go 1.15. We should probably upgrade, as 1.15 is already out of support, and 1.18 is coming (which will put 1.16 out of support as well). Anyway, that's a tangent :smile: 

Regarding the dependencies, looking into the [minimal version selection](https://go.dev/ref/mod#minimal-version-selection) algorithm used, it's not clear to me that the older versions may not end up being picked. I think the work around would be to explicitly force/fix the version number we want to use for these dependencies, instead of simply upgrading them.

I tried manually removing the older dependencies, however running `go mod tidy` again adds them back :upside_down_face: The only workaround I see is using a replace directive to force the version, however I think that may break our automatic dependency update (possibly not), and it's not really what the replace directive is meant for. So that leaves updating to 1.17 (though I think the mod pruning will only work with dependency modules which are also on 1.17), or hoping the Go dependency manager knows what it's doing and that it will use the right versions. In the end, these are still test only dependencies, so I'm not super worried about them - but it could happen with a non-test dependency in the future, so it would be good to understand it.

**Review by lenaschoenburg — APPROVED** [review]
Alright, that was an interesting tangent, thanks for explaining :+1:

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**